url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/execution/Generic_Executor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package uk.ac.leeds.ccg.andyt.generic.execution;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ *
+ * @author geoagdt
+ */
+public class Generic_Executor {
+
+    public Generic_ImageWriter i;
+    
+    public Generic_Executor() {
+    }
+
+    public Generic_Executor(Generic_ImageWriter i) {
+    }
+
+    public static void main(String[] args) {
+        new Generic_Executor().run();
+    }
+
+    public void run() {
+        int poolSize = 5;
+        ExecutorService executorService = Executors.newFixedThreadPool(5);
+        for (int i = 0; i < poolSize * 2; i++) {
+            final Generic_Runnable r = new Generic_Runnable();
+            executorService.execute(new Runnable() {
+                @Override
+                public void run() {
+                    r.run();
+                }
+            });
+        }
+        executorService.shutdown();
+        try {
+            // wait for them to finish for up to one minute.
+            executorService.awaitTermination(1, TimeUnit.MINUTES);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Check the return value of `awaitTermination()` to determine if the operation timed out while waiting for other threads to stop executing. Handle when `awaitTermination()` times out. See the following diffs on GitHub that make this code change: [diff1](https://github.com/voldemort/voldemort/commit/e44ec88c57c1293a5b71598e0ad0fed403387744#diff-c220173cebaeedd4ef7bad06e679a0938710546ed77afac1d37a3ddf25e30e97L533), [diff2](https://github.com/utPLSQL/utPLSQL-cli/commit/079e021a747fe3829f868ad08c95a6eef2cc2f1c#diff-8a4bc1fbdd325f803130235d0cc70d6ae0d4b491b7440263c4f8a55452fd45eeL204)

Learn more: [ExecutorService.awaitTermination in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html#awaitTermination-long-java.util.concurrent.TimeUnit-)